crypto/provider_core.c

```diff
@@ -722,16 +756,18 @@ int ossl_provider_activate(OSSL_PROVIDER *prov, int retain_fallbacks)
             prov->store->use_fallbacks = 0;
             CRYPTO_THREAD_unlock(prov->store->lock);
         }
-        return 1;
+        return count == 1 ? provider_flush_store_cache(prov) : 1;
```

Comment:
I suppose it does not matter that the count might not reflect the real current value.

Reply:
In this case it shouldn't, the provider won't go deactivated during this call but it could be activated again but that won't flush since the count would be 2.

I am worried that there are race conditions in the provider/store/fetch code.  There are a lot of different locks involved and after any unlock there is window where things might change.  For the moment, I'll call it a bout of _paranoia_.